### DISCUSSION TITLE
Model connections error

### DISCUSSION THREAD

**Discussion by quadrastreet** [leafsphp/leaf#62] [General]
Getting this error when using modelname::all():
```

Type: ErrorException
'
Code: 8
Message: Undefined index: connections
File: /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/Database.php
Line: 28
Trace
#0 /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/Database.php(28): Leaf\Exception\General::handleErrors(8, 'Undefined index...', '/home/siteli...', 28, Array)
#1 /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/Model.php(5): Leaf\Database::connect()
#2 /home/sitelii/api.sitelii.com/sitelii-data/vendor/composer/ClassLoader.php(444): include('/home/siteli...')
#3 /home/sitelii/api.sitelii.com/sitelii-data/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('/home/siteli...')
#4 [internal function]: Composer\Autoload\ClassLoader->loadClass('Leaf\\Model')
#5 /home/sitelii/api.sitelii.com/sitelii-data/App/Models/Model.php(9): spl_autoload_call('Leaf\\Model')
#6 /home/sitelii/api.sitelii.com/sitelii-data/Config/bootstrap.php(7): require('/home/siteli...')
#7 [internal function]: {closure}('App\\Models\\Mode...')
#8 /home/sitelii/api.sitelii.com/sitelii-data/App/Models/PredictorUser.php(4): spl_autoload_call('App\\Models\\Mode...')
#9 /home/sitelii/api.sitelii.com/sitelii-data/Config/bootstrap.php(7): require('/home/siteli...')
#10 [internal function]: {closure}('App\\Models\\Pred...')
#11 /home/sitelii/api.sitelii.com/sitelii-data/App/Controllers/PredictorsController.php(14): spl_autoload_call('App\\Models\\Pred...')
#12 [internal function]: App\Controllers\PredictorsController->users()
#13 /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/Router.php(723): call_user_func_array(Array, Array)
#14 /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/Router.php(689): Leaf\Router::invoke('\\App\\Controller...', Array)
#15 /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/Router.php(555): Leaf\Router::handle(Array, true)
#16 /home/sitelii/api.sitelii.com/sitelii-data/vendor/leafs/leaf/src/App.php(668): Leaf\Router::run(NULL)
#17 /home/sitelii/api.sitelii.com/sitelii-data/index.php(98): Leaf\App->run()
#18 {main}

```
I know that it's actually able to connect to the database, as when I have the wrong db user password, etc., I get a connection error.

I feel like I need to trigger some sort of connection, but it looks like Leaf API is already triggering a connection using my .env.

**Comment by quadrastreet** [comment]
I managed to solve this by adding the following to my index.php:
```

$db_connection = getenv("DB_CONNECTION");
$db_host = getenv("DB_HOST");
$db_port = getenv("DB_PORT");
$db_database = getenv("DB_DATABASE");
$db_username = getenv("DB_USERNAME");
$db_password = getenv("DB_PASSWORD");

$db_config = array('connections'=>
        array('mysql' => array(
                'host'    => $db_host,
                'driver'    => $db_connection,
                'database'  => $db_database,
                'username'  => $db_username,
                'password'  => $db_password,
                'charset'   => 'utf8',
                'collation' => 'utf8_unicode_ci',
                'prefix'    => '',
        )
));
Leaf\Database::config($db_config);

```

**Reply by mychidarko** [reply]
Hi @quadrastreet, I inspected this issue after seeing this discussion and it turns out that the `.env` variables were not being loaded correctly due to an unknown reason. This how however been fixed, so you can create your leaf API and MVC apps just as described in the docs without having to tweak anything, thanks🙏🏽

**Reply by mychidarko** [reply]
You might want to open an issue instead of a discussion next time though or just add to the [forum](https://github.com/leafsphp/leaf/discussions/57).

**Reply by quadrastreet** [reply]
Thanks - I'll update my project.

**Reply by quadrastreet** [reply]
Sorry... I'm new to composer, and I'm really not sure how to update leafapi using composer. I'll dig through some forums, but any help you can give would be great.

**Reply by quadrastreet** [reply]
I managed to download the needed files from github and merge my updated files (such as App/Routes/index.php), but I'd love to know the real way to implement new versions of Leaf or LeafAPI.

BTW: your fixes to the LeafAPI code fixed my issues, and I'll know next time to submit an issue.

**Reply by mychidarko** [reply]
> Sorry... I'm new to composer, and I'm really not sure how to update leafapi using composer. I'll dig through some forums, but any help you can give would be great.

This might be a little roundabout, but you can simply generate a new leaf API app and then to copy your unique application files into the new application. This would include controllers, routes, models, commands, assets, and other code specific to your application.

Alternatively, you can simply check the changelog for updated files/dependencies and apply them manually. We'll think up a better way  sometime🚀

**Reply by quadrastreet** [reply]
Sounds good; that's close to what I ended up doing.  (BTW: loving Leaf so far!)

**Reply by mychidarko** [reply]
I'm glad you're enjoying leaf🔥
